bokeh/util/logconfig.py

```diff
@@ -18,9 +18,36 @@
 from __future__ import absolute_import
 
 import logging
+import sys
 
 from ..settings import settings
 
 level = settings.py_log_level()
+bokeh_logger = logging.getLogger('bokeh')
+root_logger = logging.getLogger()
+
 if level is not None:
-    logging.basicConfig(level=level)
+    bokeh_logger.setLevel(level)
+
+if not (root_logger.handlers or bokeh_logger.handlers):
+    # No handlers configured => at least add a printer to sys.stderr for
```

Comment:
Note that `basicConfig` *also* configures a default handler (as part of other things), but it does so for the entire logging tree starting at the root logger, which is what we want to avoid in a library.

(honestly, I think this part of the logging API is badly designed and easily leads to misuse of the API, since it makes doing the right thing much harder than doing the wrong easy thing)